app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -124,6 +132,18 @@ def handle_remember_device_preference(remember_device_preference)
     save_remember_device_preference(remember_device_preference)
   end
 
+  def mfa_selection_attempt_count(increment_auth_method)
+    user_session[:mfa_attempts] ||= {}
+    if increment_auth_method == 'personal_key'
+      increment_auth_method = increment_auth_method.gsub(
+        'personal_key',
+        'personal-key',
+      )
+    end
```

Comment:
Continuing the previous discussion about trying to exempt from SessionEncryptor errors, the suggestion to explore `pii_like_keypaths` might not work so well since it's not implemented by default, but moreso because SessionEncryptor is concerned with the session and not the log detail.

Instead, I wonder if we could create a set of "safe" paths, following the existing pattern of [the `SENSITIVE_PATHS` in `SessionEncryptor`](https://github.com/18F/identity-idp/blob/9951895913dd9d5784d74731a5ae672f425481b8/lib/session_encryptor.rb#L22-L33), which would be exempt from the default error handling.

I played around with this locally and this seemed to work okay:

```rb
class SessionEncryptor
  # ...

  # Paths which would trigger sensitive key detection, but which don't contain PII
  SAFE_PATHS = [
    ['warden.user.user.session', 'mfa_attempts', 'personal_key'],
  ].freeze

  # ...

  def alert_or_raise_if_contains_sensitive_keys!(hash)
    except_nested(hash, SAFE_PATHS).deep_transform_keys do |key|
      # ...
    end
  end

  def except_nested(hash, key_paths)
    hash = hash.deep_dup
    key_paths.each do |key_path|
      nested_hash = hash.dig(*key_path[..-2])
      nested_hash&.delete!(key_path.last)
    end
    hash
  end
end
```

```rb
    it 'does not raise if exempted PII key appears' do
      session = { 'warden.user.user.session' => { 'mfa_attempts' => { 'personal_key' => 1 } } }
      session = session.deep_transform_values(&:freeze)

      expect { subject.dump(session) }.not_to raise_error(SessionEncryptor::SensitiveKeyError)
    end

    it 'maintains all values from the original session' do
      session = {
        'warden.user.user.session' => {
          'idv_new' => { 'nested' => {} },
          'mfa_attempts' => { 'personal_key' => 1 },
        },
      }

      dumped = subject.dump(session.deep_dup)
      loaded = subject.load(dumped)

      expect(loaded).to eq(session)
    end
```

It's starting to get a bit complex and I'd have some reservations about the performance impact of extra logic to deep-duplicate the session hash for every request on a session that's very unlikely to ever contain the value we care about. But I also don't feel very good about the renaming approach, since it seems to be intentionally circumventing the intent of the session encryptor in identifying a "personal key" key within the session.

Reply:
Another thought, if we're okay to just increment the count of attempts for the current method but want to ensure that we reset when switching to another method, we could track it something like...

```
user_session = {
  mfa_attempts: { attempts: 0, auth_method: 'personal_key' }
}
```

...then make sure to reset attempts if the auth method is different from what's in the session

```rb
user_session[:mfa_attempts][:attempts] = 0 if user_session[:mfa_attempts][:auth_method] != auth_method
```